src/operation/aggregate/mod.rs

```diff
@@ -54,6 +54,8 @@ impl Operation for Aggregate {
             if let Ok(cursor_doc) = body.get_document_mut("cursor") {
                 cursor_doc.remove("batchSize");
             }
+        } else {
+            body.remove("writeConcern");
```

Comment:
I decided just removing any specified writeConcern was the easiest thing to do here. This will prevent the driver from erroring when a user passes a write concern into a read aggregation via `Collection::aggregate`, but I didn't think that was much of a concern.

Reply:
What do you think of instead changing `count_documents` to manually create and execute an aggregation operation rather than going through `Collection::aggregate`? I think that should fix the issue without causing any changes to the behavior of other aggregate operations.

Reply:
Good point, that does seem to be a better solution, updated.

In fact, now that we're considering doing wrapping operations (e.g. for `watch`), this probably makes sense as its own operation wrapping `Aggregate`. Should I go ahead and do that as well?

Reply:
Yep, sounds good!

Reply:
Done. I also noticed that `CountOptions` didn't have read concern or selection_criteria, so I added those options as well.